Lib/python/std_optional.i

```diff
@@ -0,0 +1,164 @@
+// Purpose: SWIG interface file for std::optional<> type.
+//
+// This file defines macros for handling optional types in SWIG.
+//
+// Usage:
+//
+// %optional(TYPE):
+//    This macro is used to bind structs and classes to wrapped python interfaces returning None if the optional is empty.
+//    Example:
+//    %optional(MyStruct)
+//    This will generate code to handle std::optional<MyStruct>.
+//
+// The SWIG_STD_OPTIONAL_DEFAULT_TYPES macro is used to define a set of default types that are supported by SWIG when working with std::optional.
+// This macro ensures that common types such as int, double, and std::string are automatically recognized and handled by SWIG when generating bindings.
+// 
+
+%{
+    #include <optional>
+%}
+
+%include <stdint.i>
+%include <std_string.i>
+
+namespace std {
+  template<typename T> class optional {};
+}
+
+%define %optional(TYPE)
+
+%naturalvar std::optional< TYPE >;
+
+// CHECKME: when is this used
+%typemap(typecheck, precedence=SWIG_TYPECHECK_POINTER) std::optional< TYPE >, std::optional< TYPE > const & (std::optional< TYPE > tmp_ov) %{

```

Comment:
`typecheck` typemaps are used to dispatch overloaded functions/methods - see the manual for full details.